admin-api-server/src/nft/service/file/file.service.ts

```diff
@@ -0,0 +1,460 @@
+          );
+          thumbnail = await this.#resizedImg({
+            buffer: this.artifact.buffer,
+            imageSize: {
+              width: thumbnailDimensions.width,
+              height: thumbnailDimensions.height,
+            },
+            type: 'thumbnail',
+            extension: artifactExtension,
+          });
+        }
+      }
+
+      if (display && thumbnail) {
+        addedFiles.push(display);
+        addedFiles.push(thumbnail);
+      }
+    } else if (displayMissing) {
+      const artifactDimensions = sizeOf(this.artifact.buffer) as Dimension;
+      let display: File | undefined;
+      if (artifactDimensions.width < 350 && artifactDimensions.height < 350) {
+        display = Object.assign({}, this.artifact);
+        display.originalname = 'display';
+      } else {
+        const displayDimensions = this.scaledDimensions(
+          artifactDimensions.width,
+          artifactDimensions.height,
+          350,
+          350,
+        );
+        const artifactExtension = mime.getExtension(this.artifact.mimetype);
+        if (artifactExtension) {
+          display = await this.#resizedImg({
+            buffer: this.artifact.buffer,
+            imageSize: {
+              width: displayDimensions.width,
+              height: displayDimensions.height,
+            },
+            type: 'display',
+            extension: artifactExtension,
+          });
+        }
+      }
+
+      if (display) {
+        addedFiles.push(display);
+      }
+    } else if (thumbnailMissing) {
+      const display = this.allFiles.find(
+        (file) => file.originalname === 'display',
+      ) as File;
+      const displayDimensions = sizeOf(display.buffer) as Dimension;
+
+      let thumbnail: File | undefined;
+      if (displayDimensions.width < 350 && displayDimensions.height < 350) {
+        thumbnail = Object.assign({}, display);
+        thumbnail.originalname = 'thumbnail';
+      } else {
+        const displayDimensions = sizeOf(display.buffer) as Dimension;
+        const thumbnailDimensions = this.scaledDimensions(
+          displayDimensions.width,
+          displayDimensions.height,
+          250,
+          250,
+        );
+        const displayExtension = mime.getExtension(display.mimetype);
+        if (displayExtension) {
+          thumbnail = await this.#resizedImg({
+            buffer: display.buffer,
+            imageSize: {
+              width: thumbnailDimensions.width,
+              height: thumbnailDimensions.height,
+            },
+            type: 'thumbnail',
+            extension: displayExtension,
+          });
+        }
+      }
+
+      if (thumbnail) {
+        addedFiles.push(thumbnail);
+      }
+    }
+    return addedFiles;
+  }
+
+  async #imageFromVideo({
+    videoFile,
+    typeOfImage,
+    imageSize,
+  }: ImageFromVideoParams): Promise<File | undefined> {
+    const videoExtension = mime.getExtension(videoFile.mimetype);
+    if (videoExtension) {
```

Comment:
I always prefer to reduce the size of `if / else` branches, it makes code significantly easier to read because a smaller sequence of statements must be read to understand the scope of a particular special execution branch.

Usually it's actually better to deal with the bad case, and let the main execution branch scope stay on the main scope. Ie here:

```
if (!videoExtension) {
  // I think a log of why nothing got generated here should be included
  return;
}
```

Reply:
Good thought. By habit I usually funnel everything down to one last return. A quick search here https://softwareengineering.stackexchange.com/questions/118703/where-did-the-notion-of-one-return-only-come-from showed me that I am wrong and that your approach makes more sense